tests/jeos/rpi_bluetooth.pm

```diff
@@ -0,0 +1,45 @@
+# SUSE's openQA tests
+#
+# Copyright 2015-2021 SUSE LLC
+# SPDX-License-Identifier: FSFAP
+
+# Summary: Check RPi Wifi
+# Maintainer: qsf-u team <qa-core@suse.de>
+
+use base "opensusebasetest";
+use strict;
+use warnings;
+use testapi;
+use Utils::Architectures;
+use Utils::Backends;
+use utils;
+
+# This requires a discoverable bluetooth device in range:
+#===========================
+#/etc/bluetooth/main.conf:
+#[Policy]
+#AutoEnable=true
+#[General]
+#DiscoverableTimeout = 0
+#Name = openQA-worker
+#===========================
+
+sub run {
+    my ($self) = @_;
+
+    zypper_call "in bluez";
+    systemctl "start bluetooth";
+    systemctl "status bluetooth";
+    assert_script_run "rfkill list";
+    assert_script_run "bluetoothctl show";
+    assert_script_run '(echo "power on"; sleep 5; echo "scan on"; sleep 15; echo "devices") | bluetoothctl | tee /dev/stderr | grep openQA-worker';
```

Comment:
You are using `sleep()` or are simulating a sleep-like functionality. This is introducing the risk of either unnecessarily wasting time or just shifting race conditions and making them even harder to investigate. You *can* use sleep but there are only very limited cases where is the best solution (e.g. yield threads in low-level system code). Here you have the classical problem in software design of [Synchronization](https://en.wikipedia.org/wiki/Synchronization_(computer_science)) . sleep should not be used in place of proper synchronization methods. https://blogs.msmvps.com/peterritchie/2007/04/26/thread-sleep-is-a-sign-of-a-poorly-designed-program/ is one of multiple references that explains this: "The original problem is likely a timing/synchronization issue, ignoring it by hiding it with Thread.Sleep is only going to delay the problem and make it occur in random, hard to reproduce ways." And if the problem does not occur at least time is wasted. https://testing.googleblog.com/2021/03/test-flakiness-one-of-main-challenges.html states that as well: "Do NOT add arbitrary delays as these can become flaky again over time and slow down the test unnecessarily."

We are waiting because some condition changes some time ... keyword(s) is/are *some time*. It is important to know the exact condition that the code should wait for and check for that instead of delaying program execution for an arbitrary amount of time. For openQA tests in general just try to do what you as a human user would also do. You would not look on your watch waiting for exactly 100s before you look at the screen again, right? :)

See https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/CONTRIBUTING.md#coding-style for more details

Reply:
I know you don't like `sleep()` and I understand why, but this doesn't help if the SUT didn't get enough time to scan for bluetooth devices.